portico: Update screenshots for "for/X" pages .

roanster007 opened the pull request:
<!-- Describe your pull request here.-->
This PR updates screenshots for the "for/X"  pages mentioned in #30128.

Fixes #30128

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
<table>
<tr>
<td>
/for/education</td>
<td>





<img width="953" alt="image" src="https://github.com/zulip/zulip/assets/97049524/6c853395-cc32-499e-9f8a-25b0073a8f56">
</td>
<td>



<img width="953" alt="image" src="https://github.com/zulip/zulip/assets/97049524/c3a63489-c098-4ebd-9b30-129eca5dc74f">
</td>
</tr>

<tr>
<td>
/for/events</td>
<td>




<img width="953" alt="image" src="https://github.com/zulip/zulip/assets/97049524/cc33f68d-a392-47f1-ac85-6301c6f2976a">


</td>
<td>


<img width="953" alt="image" src="https://github.com/zulip/zulip/assets/97049524/d5504cf9-0a0f-4c71-a382-3bbf1b4ff336">
</td>


</tr>

<tr>
<td>
/for/business or open-source</td>
<td>





<img width="953" alt="image" src="https://github.com/zulip/zulip/assets/97049524/893a4ba1-a741-4595-8a8c-10d193e94150">
</td>
<td>



<img width="953" alt="image" src="https://github.com/zulip/zulip/assets/97049524/a9049ef0-4160-4498-88e5-20b35bc0c6bb">
</td>
</tr>

<tr>
<td>
/for/research</td>
<td>



<img width="953" alt="image" src="https://github.com/zulip/zulip/assets/97049524/be483517-baab-4ed9-b833-de8d308f500b">
</td>
<td>



<img width="953" alt="image" src="https://github.com/zulip/zulip/assets/97049524/5faa009a-ba93-4b6b-b8fc-4d1e4dc4e154">
</td>
</tr>

</table>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

roanster007 commented:
@alya updated all the screenshots (with avatars) mentioned in the issue.

alya commented:
Thanks, and sorry for the slow review!

1. The font size was already smaller than ideal, and is even tinier now. I think we should just trim the screenshots more: use a narrow window, and end just to the right of the message content (cutting off the timestamps).

roanster007 commented:
The images won't be updated in the respective pages since previous figma images combined images of 2 different topics as a single image. So, I had to keep the same image name/path, and add "_1" and "_2" suffixes to distinguish them.

alya commented:
I'm not sure what you mean the images being manually updated on the page. All the code for these pages is in the zulip/zulip repo, and is updated in PRs like everything else.

roanster007 commented:
@alya I updated the screenshots, narrowing the web page (to appear in similar width of existing screenshots), and updated the corresponding pages to reflect the changes.

alya commented:
Thanks! Please add screenshots of the resulting pages to the PR description.

roanster007 commented:
I guess we need to modify stream/topics names for some screenshots since they appear too long after narrowing.

alya commented:
Let's use the strategy we had before of cutting off the timestamps in the screenshots. They just kind of add noise in any case. Then we'll have more space for the channel/topic names.

roanster007 commented:
@alya I narrowed the window size, removed the timestamp column, and  updated the pr and screenshots.

alya commented:
Thanks! Hm, I think these would look better if we could include the left edge. I think it won't cost that much horizontal space... Maybe try one page and post in #design, @-mentioning me? We should try to get this settled, without getting delayed by how often I work through updated PRs.

andersk commented:
You’ve incorrectly changed the equation in `events/message_formatting_day.png` by adding a “+” that shouldn’t be there. Also, it was intended to demonstrate display math (```` ```math ````), not inline math (`$$`).

andersk commented:
Your code blocks in `companies/message-formatting-top.png` and `education/message_formatting_day_1.png` start with a leading space character that shouldn’t be there.

andersk commented:
`education/interactive_messaging_day_2.png` shows an unfortunate wrapping bug between “6:30 PM” and “!”. Part of this is #30515, but that doesn’t fix it by itself; we also need to avoid styling `<time>` with `display: inline-block`.

andersk commented:
`education/message_formatting_day_2.png` shows some extra vertical space between Zoe’s name and her message. This seems to be a KaTeX CSS bug where `.katex-mathml` inflates the line-height even though it’s visually hidden in various ways.

alya commented:
Are we accidentally applying the right-side fading to the channels/topics image too? It doesn't make sense to do that on that one (which is not cut off in the same way, and has a border).
![Screenshot 2024-06-21 at 14 22 34@2x](https://github.com/zulip/zulip/assets/2090066/06da08d4-5ebd-4abb-9d64-95b75d0f50f7)

alya commented:
Looking close otherwise!

roanster007 commented:
>Are we accidentally applying the right-side fading to the channels/topics image too? It doesn't make sense to do that on that one (which is not cut off in the same way, and has a border).

oops, I mixed up css class there. @alya fixed it.

roanster007 commented on tools/screenshots/thread-screenshot.js:
```diff
@@ -64,17 +64,33 @@ async function run() {
         const messageListBox = await page.$(messageListSelector);
         await page.evaluate((msg) => $(msg).removeClass("selected_message"), messageSelector);
 
+        // This is done so as to get white background while capturing screenshots.
+        const background_selectors = [".app-main", ".message-feed", ".message_header"];
+        await page.evaluate((selectors) => {
+            for (const selector of selectors) {
+                $(selector).css("background-color", "white");
+            }
+        }, background_selectors);
+
```
This is to make sure there is no grey area when the messsagebox screenshot is captured (since it is curved, despite however close we try to capture the screenshot)

timabbott commented on tools/screenshots/for-events.json:
```diff
@@ -0,0 +1,84 @@
+        "color": "#C6A8AD",
+        "topic": "slides and presentation",
+        "users": {
+            "Dal Kim": "static/images/user_avatars/DalKim.jpg",
```
Since the avatars are only used for this tool, should we have them live in `tools/screenshots/user_avatars/` instead? I feel like that scopes them more nicely than putting them in `static/images`, which might make it look like they would be used in the actual website directly.

Also, it might be a good idea to just have the basename, `DalKim.jpg`, be in these data structures, and the code itself be responsible for adding the path; that would greatly simplify changing where the avatars live in the user.

timabbott replied:
I guess I'm also not sure we need the name=>avatar mappings be in these individual JSON files at all; could they just be in `thread-screenshot.js`, if the avatars for each user are always the same?

And then this structure could be just `"Dal Kim": {},`, with the idea being the `{}` would be a space for user configuration options in the future (like if we need to set a role or something) but would be just discarded by the script for now.

roanster007 replied:
ya, that would make it really clean.

timabbott commented on web/styles/portico/landing_page.css:
```diff
@@ -2284,6 +2285,30 @@ button {
+                transparent 100%
+            );
+        }
+    }
```
What does this `starred` class mean? I feel like the name could be more clear/specific.

roanster007 replied:
`starred` is class used when screenshot has starred message --> reduces the fading effect of image so that the star is visible.

I updated it to `message-starred` to be more specific.

timabbott replied:
OK, we should add a comment documenting this.

roanster007 replied:
added:
```    
/* message-starred is used when the message screenshot captured using
the script "generate-user-message-screenshot" contains starred message.
This is done to reduce the fading effect on the image caused by
".message-screenshot" class so that the star is visible. */
```

roanster007 commented:
@timabbott this is ready for another review.

timabbott commented on static/images/landing-page/education/interactive_messaging_day_1.png:
This commit message contains `Partly Fixes #30128`, which will close the issue when this commit is merged, because GitHub's regexes are really dumb.

roanster007 replied:
Updated!

timabbott commented:
Looks great, merged, thanks for all the work on this @roanster007!
